### PR TITLE
Refactor rolling paper to use customizable reagent holder

### PR DESCRIPTION
**Pull request by jacklecroy**
:cl: That REALLY Good Soda Flavor
refactor: Rolling paper will now use the customizable reagent holder component. This cleans up the code and expands the possibilities for using the component by adding support for dried items.
/:cl:

**Details:**
The component can accept dryable items and food items now. Any instance of the component can be set to use dryable items, just like normally food items are used. To add the dryable item, it's the same as with food. You attack with the item in hand. I am currently thinking about more uses for dryable items and also more ways the component can be used.

**Why it's good:**
Expands the potential for this component and reduces the amount of code on the rolling paper.

### CODE REVIEW

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -893,25 +893,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	icon_state = "cig_paper"
 	w_class = WEIGHT_CLASS_TINY
 
-/obj/item/rollingpaper/afterattack(atom/target, mob/user, proximity)
+/obj/item/rollingpaper/Initialize()
```
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:896]:
```suggestion
/obj/item/rollingpaper/Initialize(mapload)
```
#61623 will live on

**Comment by Watermelon914** [comment]
can you update your description to explain what the difference between the old and new code is?

**Comment by jacklecroy** [comment]
> can you update your description to explain what the difference between the old and new code is? It's really not descriptive enough.

It doesn't change anything in game. It's just a refactor to use the component now.

**Comment by Watermelon914** [comment]
> > It still needs some description as to why you are doing this change or what it changes by doing this (not in game).
> 
> OK I've added more, is that better?

I meant in your description not in your changelog.

**Comment by jacklecroy** [comment]
> I meant in your description not in your changelog.
Ok I have added more detail

**Pull request merged**